breakpad-symbols/src/sym_file/parser.rs

```diff
@@ -27,265 +35,282 @@ enum Line {
+    let (input, (_, url)) = tuple((
+        terminated(tag("INFO URL"), multispace1),
+        terminated(map_res(not_my_eol, str::from_utf8), my_eol),
+    ))(input)?;
+    Ok((input, Info::Url(url.to_string())))
+}
 
 // Matches other INFO records.
-named!(
-    info_line,
-    chain!(
-        tag!("INFO") ~
-        space ~
-        x: not_line_ending ~
-        my_eol,
-          ||{ x }
-    )
-);
+fn info_line(input: &[u8]) -> IResult<&[u8], &[u8]> {
+    let (input, (_, info)) = tuple((
+        terminated(tag("INFO"), multispace1),
+        terminated(not_my_eol, my_eol),
+    ))(input)?;
+    Ok((input, info))
+}
 
 // Matches a FILE record.
-named!(file_line<&[u8], (u32, String)>,
-  chain!(
-    tag!("FILE") ~
-    space ~
-    id: decimal_u32 ~
-    space ~
-    filename: map_res!(not_line_ending, str::from_utf8) ~
-    my_eol ,
-      ||{ (id, filename.to_string()) }
-));
+fn file_line(input: &[u8]) -> IResult<&[u8], (u32, String)> {
+    let (input, (_, id, filename)) = tuple((
+        terminated(tag("FILE"), multispace1),
+        terminated(decimal_u32, multispace1),
+        terminated(map_res(not_my_eol, str::from_utf8), my_eol),
+    ))(input)?;
+    Ok((input, (id, filename.to_string())))
+}
 
 // Matches a PUBLIC record.
-named!(public_line<&[u8], PublicSymbol>,
-  chain!(
-    tag!("PUBLIC") ~
-    preceded!(space, tag!("m"))? ~
-    space ~
-    address: hex_str_u64 ~
-    space ~
-    parameter_size: hex_u32 ~
-    space ~
-    name: map_res!(not_line_ending, str::from_utf8) ~
-    my_eol ,
-      || {
-          PublicSymbol {
-              address,
-              parameter_size,
-              name: name.to_string()
-          }
-      }
-));
+fn public_line(input: &[u8]) -> IResult<&[u8], PublicSymbol> {
+    let (input, (_, _multiple, _, address, parameter_size, name)) = tuple((
+        tag("PUBLIC"),
+        opt(tag(" m")),
+        multispace1,
+        terminated(hex_str_u64, multispace1),
+        terminated(hex_u32, multispace1),
+        terminated(map_res(not_my_eol, str::from_utf8), my_eol),
+    ))(input)?;
+    Ok((
+        input,
+        PublicSymbol {
+            address,
+            parameter_size,
+            name: name.to_string(),
+        },
+    ))
+}
 
 // Matches line data after a FUNC record.
-named!(func_line_data<&[u8], SourceLine>,
-  chain!(
-    address: hex_str_u64 ~
-    space ~
-    size: hex_u32 ~
-    space ~
-    line: decimal_u32 ~
-    space ~
-    filenum: decimal_u32 ~
-    my_eol ,
-      || {
-          SourceLine {
-              address,
-              size,
-              file: filenum,
-              line,
-          }
-      }
-));
+fn func_line_data(input: &[u8]) -> IResult<&[u8], SourceLine> {
+    let (input, (address, size, line, file)) = tuple((
+        terminated(hex_str_u64, multispace1),
+        terminated(hex_u32, multispace1),
+        terminated(decimal_u32, multispace1),
+        terminated(decimal_u32, my_eol),
```

Comment:
I believe you're preserving semantics but it's interesting that we'll freak out if there's trailing whitespace